src/rate.rs

```diff
@@ -0,0 +1,622 @@
+}
+
+// Rate - Rate = Rate (to make shorthands work, until const_generics_defaults is
+// stabilized)
+impl<const NOM: u32, const DENOM: u32> ops::Sub<Rate<u32, NOM, DENOM>> for Rate<u64, NOM, DENOM> {
+    type Output = Rate<u64, NOM, DENOM>;
+
+    #[inline]
+    fn sub(self, other: Rate<u32, NOM, DENOM>) -> Self::Output {
+        if let Some(v) = self.checked_sub(Rate::<u64, NOM, DENOM>::from_ticks(other.ticks() as u64))
+        {
+            v
+        } else {
+            panic!("Sub failed!");
+        }
+    }
+}
+
+// Rate + Rate = Rate (to make shorthands work, until const_generics_defaults is
+// stabilized)
+impl<const NOM: u32, const DENOM: u32> ops::Add<Rate<u32, NOM, DENOM>> for Rate<u64, NOM, DENOM> {
+    type Output = Rate<u64, NOM, DENOM>;
+
+    #[inline]
+    fn add(self, other: Rate<u32, NOM, DENOM>) -> Self::Output {
+        if let Some(v) = self.checked_add(Rate::<u64, NOM, DENOM>::from_ticks(other.ticks() as u64))
+        {
+            v
+        } else {
+            panic!("Add failed!");
+        }
+    }
+}
+
+impl<const L_NOM: u32, const L_DENOM: u32, const R_NOM: u32, const R_DENOM: u32>
+    PartialOrd<Rate<u32, R_NOM, R_DENOM>> for Rate<u64, L_NOM, L_DENOM>
+{
+    #[inline]
+    fn partial_cmp(&self, other: &Rate<u32, R_NOM, R_DENOM>) -> Option<Ordering> {
+        self.partial_cmp(&Rate::<u64, R_NOM, R_DENOM>::from_ticks(
+            other.ticks() as u64
+        ))
+    }
+}
+
+impl<const L_NOM: u32, const L_DENOM: u32, const R_NOM: u32, const R_DENOM: u32>
+    PartialEq<Rate<u32, R_NOM, R_DENOM>> for Rate<u64, L_NOM, L_DENOM>
+{
+    #[inline]
+    fn eq(&self, other: &Rate<u32, R_NOM, R_DENOM>) -> bool {
+        self.eq(&Rate::<u64, R_NOM, R_DENOM>::from_ticks(
+            other.ticks() as u64
+        ))
+    }
+}
+
+impl<const L_NOM: u32, const L_DENOM: u32, const R_NOM: u32, const R_DENOM: u32>
+    PartialOrd<Rate<u64, R_NOM, R_DENOM>> for Rate<u32, L_NOM, L_DENOM>
+{
+    #[inline]
+    fn partial_cmp(&self, other: &Rate<u64, R_NOM, R_DENOM>) -> Option<Ordering> {
+        Rate::<u64, L_NOM, L_DENOM>::from_ticks(self.ticks as u64).partial_cmp(other)
+    }
+}
+
+impl<const L_NOM: u32, const L_DENOM: u32, const R_NOM: u32, const R_DENOM: u32>
+    PartialEq<Rate<u64, R_NOM, R_DENOM>> for Rate<u32, L_NOM, L_DENOM>
+{
+    #[inline]
+    fn eq(&self, other: &Rate<u64, R_NOM, R_DENOM>) -> bool {
+        Rate::<u64, L_NOM, L_DENOM>::from_ticks(self.ticks as u64).eq(other)
+    }
+}
+
+/// Extension trait for simple short-hands for u32 Rate
+pub trait ExtU32 {
+    /// Shorthand for creating a duration which represents hertz.
+    fn hz<const NOM: u32, const DENOM: u32>(self) -> Rate<u32, NOM, DENOM>;
+
+    /// Shorthand for creating a duration which represents kilohertz.
+    fn khz<const NOM: u32, const DENOM: u32>(self) -> Rate<u32, NOM, DENOM>;
+
+    /// Shorthand for creating a duration which represents megahertz.
+    fn mhz<const NOM: u32, const DENOM: u32>(self) -> Rate<u32, NOM, DENOM>;
```

Comment:
Any chance these could be `Hz`, `kHz` and `MHz` to match typical conventions? `mhz` in particular is *slightly* ambiguous. I doubt anyone is actually using millihertz, but it still looks weird to me personally.

Reply:
I don't understand? I was suggesting `fn Hz(self)`, `fn kHz(self)` and `fn MHz(self)`.

Reply:
I would let SI prefix/unit/shorthand nomenclature trump Rust naming conventions. `mHz` and `MHz` are two very different things. Both are real and actual valid use-cases. Having `mhz` would be very ambiguous.

Reply:
also `ticks/from_ticks` are not very good names for rate

Reply:
Yeah, I'll update the rate names.

@burrbull what alternative would you have in mind instead of ticks?

Reply:
Actually I do not have good idea.
Maybe just `from_raw`?

Reply:
Same here, `from_raw` is a good though :)